caringcaribou/modules/xcp.py

```diff
@@ -273,14 +279,20 @@ def connect_callback_handler(msg):
             def callback_handler(msg):
                 global command_reply
                 if msg.arbitration_id == rcv_arb_id:
-                    print("{0:<23} {1}".format(cmd_desc, msg.data[0] != 0xfe))
+                    if msg.data[0] == 0xFF:
+                        print("{0:<23} AVAILABLE".format(cmd_desc))
+                    elif (msg.data[0] == 0xFE) and (msg.data[1] == 0x20):
```

Comment:
Trying to read from `msg.data[1]` will raise an `IndexError: bytearray index out of range` if `msg` only contains one byte.

A length check (e.g. `len(msg.data) > 1`) is needed before trying to read this index.

Reply:
Yes, a length check is better. I didn't put it because it was working as it is in my case